app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -124,6 +124,8 @@ def pulp3_configuration(config_class)
           config.ssl_client_key = ::Cert::Certs.ssl_client_key
           config.debugging = true
           config.logger = ::Foreman::Logging.logger('katello/pulp_rest')
+          config.username = self.setting(PULP3_FEATURE, 'username') 
+          config.password = self.setting(PULP3_FEATURE, 'password')
```

Comment:
the smart proxy has to be configured to actually send these settings: https://github.com/theforeman/smart_proxy_pulp/blob/master/lib/smart_proxy_pulp_plugin/pulp_plugin.rb#L10

otherwise i think adding this option is fine (although i don't see our installed deployments actually using it).

Reply:
Like so: https://github.com/theforeman/smart_proxy_pulp/pull/23